### PR TITLE
Add react-rails for and teaspoon-mocha front-end React components

### PR DESCRIPTION
**Pull request by graysonwright**
See:

- https://github.com/reactjs/react-rails
- https://github.com/modeset/teaspoon
- https://github.com/modeset/teaspoon/wiki/Using-Mocha

### CODE REVIEW

File: app/assets/javascripts/components/hello_world.es6.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    return <div>
+    You've clicked the link {this.state.count} times.
+    <span onClick={this.click.bind(this)}>Click me!</span>
+    </div>;
+  }
+
+  click() {
+    this.setState({count: this.state.count + 1});
+  }
+}
```
Comment (paulcsmith) [inline app/assets/javascripts/components/hello_world.es6.jsx]: Do we need this HelloWorld class or is it automatically generated and can be removed?

Comment (paulcsmith) [inline app/assets/javascripts/components/hello_world.es6.jsx]: It might be nice to leave it in though as a jumping off point

**Comment by paulcsmith** [comment]
I think this looks good!

File: app/assets/javascripts/components/hello_world.es6.jsx (hunk 1 above)
Comment (graysonwright) [inline app/assets/javascripts/components/hello_world.es6.jsx]: @paulcsmith I created it to test that everything was set up correctly. I also just added a spec for it in 9d18b8d, to verify the test framework is working.

Comment (paulcsmith) [inline app/assets/javascripts/components/hello_world.es6.jsx]: :+1: sounds good! Once we have some real code in there we can remove it. We'll just have to make sure we remember haha

**Pull request merged**